### PR TITLE
feat: Don't allow duplicate cat-groups in budget

### LINKED ISSUE #1920: [Bug]: Don't allow duplicate category group names
### Verified issue does not already exist?

- [X] I have searched and found no existing issue

### What happened?

We should not allow duplicate names for category groups

### PR DESCRIPTION
**Pull request by dhruvramdev**
Checks:
- UI Level check to check for checking duplicate group name
- DB level check for the same

Inspired from PR #1833 
Fixes #1920

### CODE REVIEW

**Comment by youngcw** [comment]
Could you match the error for categories?  That one calls out the possibility of hidden

**Comment by joel-jeremy** [comment]
This results in an internal server error notification instead of a more helpful "duplicate category group" error notification.

**Comment by youngcw** [comment]
> Can we also add the check in the mobile budget page? This results in an internal server error notification instead of a more helpful "duplicate category group" error notification in mobile budget page.

Looks like it also happens when adding a duplicate category.

**Comment by youngcw** [comment]
@joel-jeremy Would you be ok if those mobile issues where their own PR?  I think this one is fine on its own since it does match the existing behavior of the setting categories pretty well

**Comment by joel-jeremy** [comment]
I'm fine with merging this as is but that would introduce inconsistencies between desktop and mobile.

**Comment by youngcw** [comment]
@dhruvramdev This all looks good.  Im willing to merge this if you add the note about hidden, like the category check does.

File: packages/desktop-client/src/components/budget/index.tsx [hunk 1]
```diff
@@ -313,7 +313,25 @@ function BudgetInner(props: BudgetProps) {
     }
   };
 
+  const groupNameAlreadyExistsNotification = name => {
+    props.addNotification({
+      type: 'error',
+      message: `Group ‘${name}’ already exists in budget (May be Hidden)`,
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/index.tsx:319]: I think you can pass in the existing group in this method and set the message accordingly if the group is actually hidden.

**Comment by dhruvramdev** [comment]
@joel-jeremy - Not to expand the scope for this PR (all this will be done in seperate PR) but I was looking at implementing the same for MobileBudget component, what's the preferred way of reporting the error on mobile? I believe while adding a category there is a validation check with error message support but for updating, right now there is no error handling. Should we just add a notification similar to this PR or make the error message similar to add category group?

File: packages/desktop-client/src/components/budget/index.tsx [hunk 2]
```diff
@@ -313,7 +313,24 @@ function BudgetInner(props: BudgetProps) {
     }
   };
 
+  const groupNameAlreadyExistsNotification = group => {
+    props.addNotification({
+      type: 'error',
+      message: `Group ‘${group.name}’ already exists in budget ${group.hidden ? '(May be Hidden)' : ''}`,
```
Comment (joel-jeremy) [inline packages/desktop-client/src/components/budget/index.tsx:319]:
```suggestion
      message: `A ${group.hidden ? 'hidden' : ''}’ ${group.name}’ category group already exists.`,
```

File: packages/loot-core/src/server/db/index.ts [hunk 3]
```diff
@@ -304,6 +304,17 @@ export async function getCategoriesGrouped(): Promise<
...
 
 export async function insertCategoryGroup(group) {
+  // Don't allow duplicate group
+  const existingGroup = await first(
+    `SELECT id FROM category_groups WHERE UPPER(name) = ? and tombstone = 0 LIMIT 1`,
+    [group.name.toUpperCase()],
+  );
+  if (existingGroup) {
+    throw new Error(
+      `Group ‘${group.name}’ already exists in budget (May be Hidden)`,
```
Comment (joel-jeremy) [inline packages/loot-core/src/server/db/index.ts:314]: Can we also use the same error message here?

**Comment by joel-jeremy** [comment]
> @joel-jeremy - Not to expand the scope for this PR (all this will be done in seperate PR) but I was looking at implementing the same for MobileBudget component, what's the preferred way of reporting the error on mobile? I believe while adding a category there is a validation check with error message support but for updating, right now there is no error handling. Should we just add a notification similar to this PR or make the error message similar to add category group?

IMO the error feedback should be same between mobile and desktop. They should have the same error notification and message.

**Review by joel-jeremy — APPROVED** [review]
Thank you! LGTM

**Pull request merged**